Storage/tests/Unit/StorageObjectTest.php

```diff
@@ -512,30 +498,94 @@ public function testDownloadAsStreamWithExtraOptions()
         $bucket = 'bucket';
         $object = self::OBJECT;
         $stream = Utils::streamFor($string = 'abcdefg');
-        $this->connection->downloadObject([
-            'bucket' => $bucket,
-            'object' => $object,
-            'restOptions' => [
-                'headers' => [
-                    'x-goog-encryption-algorithm' => 'AES256',
-                    'x-goog-encryption-key' => $key,
-                    'x-goog-encryption-key-sha256' => $hash
-                ]
-            ]
-        ])
-            ->willReturn($stream);
+        $this->connection->downloadObject(Argument::allOf(
+            Argument::withKey('bucket'),
+            Argument::withKey('object'),
+            Argument::withKey('restOptions')
+        ))->willReturn($stream);
 
         $object = new StorageObject($this->connection->reveal(), $object, $bucket);
 
         $body = $object->downloadAsStream([
             'encryptionKey' => $key,
-            'encryptionKeySHA256' => $hash
+            'encryptionKeySHA256' => $hash,
+            'size' => $stream->getSize()
+        ]);
+
+        $this->assertInstanceOf(StreamInterface::class, $body);
+        $this->assertEquals($string, $body);
+    }
+
+    /**
+     * Checks if the `onRetryException` callable is passed
+     * when the user has supplied their own Range headers.
+     */
+    public function testDownloadAsStreamWithRangeHeaders()
+    {
+        $bucket = 'bucket';
+        $object = self::OBJECT;
+        $stream = Utils::streamFor($string = 'abcdefg');
+        $this->connection->downloadObject(Argument::allOf(
+            Argument::withKey('bucket'),
+            Argument::withKey('object')
+        ))->willReturn($stream);
+
+        // We expect no $options['onRetryException] param to be passed
+        // to the downloadObject method
+        $this->connection->downloadObject(
+            Argument::withKey('onRetryException')
+        )->shouldNotBeCalled();
+
+        $object = new StorageObject($this->connection->reveal(), $object, $bucket);
+
+        // The size param is passed just to avoid an extra call to
+        // $this->connection->getObject
```

Comment:
Are the tests using the size parameter still valid with the latest update?

Reply:
No, it isn't. Thanks for the catch.